packages/checkout/registry/index.js

```diff
@@ -0,0 +1,56 @@
+let checkoutFilters = {};
+
+/**
+ * Register filters for a specific extension.
+ *
+ * @param {string} namespace Name of the extension namespace.
+ * @param {Object} filters   Object of filters for that namespace. Each key of
+ *                           the object is the name of a filter.
+ */
+export const __experimentalRegisterCheckoutFilters = ( namespace, filters ) => {
+	checkoutFilters = {
+		...checkoutFilters,
+		[ namespace ]: filters,
+	};
+};
+
+/**
+ * Get all filters with a specific name.
+ *
+ * @param {string} filterName   Name of the filter to search for.
+ * @return {Function[]} Array of functions that are registered for that filter
+ *                      name.
+ */
+const getCheckoutFilters = ( filterName ) => {
+	const namespaces = Object.keys( checkoutFilters );
+	const filters = namespaces
+		.map( ( namespace ) => checkoutFilters[ namespace ][ filterName ] )
+		.filter( Boolean );
+	return filters;
+};
+
+/**
+ * Apply a filter.
+ *
+ * @param {string}   filterName   Name of the filter to apply.
+ * @param {any}      defaultValue Default value to filter.
+ * @param {Object}   [args]       Arguments to pass to registered functions.
+ * @param {Function} [validate]   Function that needs to return true when the
+ *                                filtered value is passed in order for the
+ *                                filter to be applied.
+ * @return {any} Filtered value.
+ */
+export const __experimentalApplyCheckoutFilter = (
+	filterName,
+	defaultValue,
+	args = {},
+	validate = () => true
+) => {
+	const filters = getCheckoutFilters( filterName );
+	let value = defaultValue;
```

Comment:
There is no enforcing that the new string includes the original or default value, is this intended?

Reply:
For this particular filter, this is intended because we allow extensions to completely replace the `Total` label with whatever they want. For other filters that change price format, for example, we will need to check that it includes the original string. The good thingk is that each filter can specify a different `validation` function.

Reply:
I guess that makes sense, it does cause a lot of noise in the component, but I think we can clean it by time.